girder/utility/plugin_utilities.py

```diff
@@ -189,40 +220,42 @@ def findAllPlugins(curConfig=None):
               'loaded.'))
         return allPlugins
-    dirs = [dir for dir in os.listdir(pluginsDir) if os.path.isdir(
-            os.path.join(pluginsDir, dir))]
-
-    for plugin in dirs:
-        data = {}
-        configJson = os.path.join(pluginsDir, plugin, 'plugin.json')
-        configYml = os.path.join(pluginsDir, plugin, 'plugin.yml')
-        if os.path.isfile(configJson):
-            with open(configJson) as conf:
-                try:
-                    data = json.load(conf)
-                except ValueError as e:
-                    print(TerminalColor.error(
-                        'ERROR: Plugin "%s": plugin.json is not valid JSON.' %
-                        plugin))
-                    print(e)
-                    continue
-        elif os.path.isfile(configYml):
-            with open(configYml) as conf:
-                try:
-                    data = yaml.safe_load(conf)
-                except yaml.YAMLError as e:
-                    print(TerminalColor.error(
-                        'ERROR: Plugin "%s": plugin.yml is not valid YAML.' %
-                        plugin))
-                    print(e)
-                    continue
-
-        allPlugins[plugin] = {
-            'name': data.get('name', plugin),
-            'description': data.get('description', ''),
-            'version': data.get('version', ''),
-            'dependencies': set(data.get('dependencies', []))
-        }
+
+    for pluginDir in pluginsDir:
+        dirs = [dir for dir in os.listdir(pluginDir) if os.path.isdir(
+        os.path.join(pluginDir, dir))]
+
+        for plugin in dirs:
+            data = {}
+            configJson = os.path.join(pluginDir, plugin, 'plugin.json')
+            configYml = os.path.join(pluginDir, plugin, 'plugin.yml')
+            if os.path.isfile(configJson):
+                with open(configJson) as conf:
+                    try:
+                        data = json.load(conf)
+                    except ValueError as e:
+                        print(TerminalColor.error(
+                            'ERROR: Plugin "%s": plugin.json is not valid JSON.' %
+                            plugin))
+                        print(e)
+                        continue
+            elif os.path.isfile(configYml):
+                with open(configYml) as conf:
+                    try:
+                        data = yaml.safe_load(conf)
+                    except yaml.YAMLError as e:
+                        print(TerminalColor.error(
+                            'ERROR: Plugin "%s": plugin.yml is not valid YAML.' %
+                            plugin))
+                        print(e)
+                        continue
+
+            allPlugins[plugin] = {
+                'name': data.get('name', plugin),
+                'description': data.get('description', ''),
+                'version': data.get('version', ''),
+                'dependencies': set(data.get('dependencies', []))
+            }
 
```

Comment:
@danlamanna pointed out that this method takes as an argument the cherrypy config (or uses the current config) and isn't easily memoized.  In addition, this is probably a textbook case of premature optimization.  With that in mind, I think we should move ahead with this PR without addressing it.

I don't think there was anything else holding this back so I'll pass it @cjh1 and @zachmullen to weigh in.